.github/pull_request_template.md

```diff
@@ -1,23 +1,69 @@
 <!--
+A few suggestions about filling out this PR
 
-Replace this comment with a description of what's being changed by this PR.
+1. Use a descriptive title for the PR.
+2. If this pull request is work in progress, create a draft PR instead of prefixing the title with WIP.
+3. Please label this PR at least one of the following labels, depending on the scope of your change:
+- type:new-feature, which adds new behavior
+- type:bug fix
+- type:enhancement, which modifies existing behavior
+- type:breaking-change
+4. Remove those recommended/optional sections if you don't need them. Only "What does this PR do" and "Checklist" are mandatory.
+5. Submit the pull request: Push your local changes to your forked copy of the repository and submit a pull request (https://help.github.com/articles/using-pull-requests).
+6. Please be patient. We might not be able to review your code as fast as we would like to, but we'll do our best to dedicate to it the attention it deserves. Your effort is much appreciated!
+-->
 
-If this PR should close an issue, please add one of the magic keywords
-(e.g. Fixes) followed by the issue number. For more info see:
-https://help.github.com/articles/closing-issues-using-keywords/
+## What does this PR do?
+<!-- _(Mandatory)_
+Replace this comment with a description of what's being changed by this PR. Please explain the WHAT: A clear and concise description of what (patterns used, algorithms implemented, design architecture, message processing, etc.)
+-->
 
-If this pull request is work in progress, create a draft PR instead of prefixing the title with WIP.
+## Checklist
+<!-- _(Mandatory)_
+List here all the items you have verified BEFORE sending this PR. Please DO NOT remove any item, striking through those that do not apply. (Just in case, strikethrough uses two tildes. ~~Scratch this.~~)
+-->
+- [ ] I have signed the [Contributor License Agreement](https://www.elastic.co/contributor-agreement/). 
+<!-- 
+Please make sure you have signed our [Contributor License Agreement](https://www.elastic.co/contributor-agreement/). We are not asking you to assign copyright to us, but to give us the right to distribute your code without restriction. We ask this of all contributors in order to assure our users of the origin and continuing existence of the code. You only need to sign the CLA once.
+-->
+- [ ] My code follows the [style guidelines of this project](CONTRIBUTING.md#java-language-formatting-guidelines)
+- [ ] I have rebased my changes on top of the latest master branch
+<!--
+Update your local repository with the most recent code from the main repo, and rebase your branch on top of the latest master branch. We prefer your initial changes to be squashed into a single commit. Later, if we ask you to make changes, add them as separate commits. This makes them easier to review. As a final step before merging we will either ask you to squash all commits yourself or we'll do it for you.
```

Comment:
```suggestion
Update your local repository with the most recent code from the main repo, and rebase your branch on top of the latest master branch. We prefer your initial changes to be squashed into a single commit. Later, if we ask you to make changes, add them as separate commits. This makes them easier to review.
```

We always squash on merge via GitHub

Reply:
This was moved from the contribution guidelines :) I'll apply your commit in the PR